Filling the `CHANGELOG.md` file (#68)
ios-course/swiftcowboys-team-project: CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file.

The project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

-  Created initial Xcode project with `MVVM` folder architecture. Added `SwiftFormat`, `SwiftLint` with rules as part of the build phase. Added `.gitignore`, this `CHANGELOG.md` and an empty `README.md` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Filled the `README` with the description and "Authors" section - [#31](https://github.com/ios-course/swiftcowboys-team-project/pull/31)
-  Fixed the bug after merging when initial information appeared in the`README.md` - [#32](https://github.com/ios-course/swiftcowboys-team-project/pull/32/files)
-  Added images from `Figma` to the assets - [#33](https://github.com/ios-course/swiftcowboys-team-project/pull/33)
-  Fixed an error while building the `Sleep tracker/Tests` target - [#34](https://github.com/ios-course/swiftcowboys-team-project/pull/34)
-  Added `Color` extension and `AppScheme` enum (with tests) according to the project color scheme - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)

> -  Created `SleepPageView` with subviews: `AlarmSetupView`, `EstimatedSleepDurationLabelView`, `NoAlarmLabelView`, `RoundedAlarmView`, `WheelTimePickerView`, according to the app design - [#39](https://github.com/ios-course/swiftcowboys-team-project/pull/39)

Reviewer: Did we create a `SleepPageView`?
I thought we have created `SleepSetupView` there.

Author: you're right. Fixed

-  Created `DuringSleepView` according to the app design - [#40](https://github.com/ios-course/swiftcowboys-team-project/pull/40)
-  Added `Image` extension and `AppImage` enum (with tests) with images according to project assets - [#51](https://github.com/ios-course/swiftcowboys-team-project/pull/51)

> -  Created `RoundedRectangleView` according to the app design - [#55](https://github.com/ios-course/swiftcowboys-team-project/pull/55)

Reviewer: nit:

```suggestion
-  Created `RoundedRectangleView` to show X/to allow Y/etc - [#55](https://github.com/ios-course/swiftcowboys-team-project/pull/55)
```

I'd suggest to have something like that.

Author: I think in this lin there is no sense at all to write it. But I double-checked the rest of the lines. Thanks

Reviewer: It's up to you, but then it will be much easier to write release notes.

-  Added `RELEASENOTES.md` file with initial structure - [#58](https://github.com/ios-course/swiftcowboys-team-project/pull/58)
-  Removed all images with 1x scale from project assets - [#61](https://github.com/ios-course/swiftcowboys-team-project/pull/61)
-  Removed spaces from all project folder/file names - [#62](https://github.com/ios-course/swiftcowboys-team-project/pull/62)

> -  Added accidentally uncommitted changes for `SleepSetupView` - [#67](https://github.com/ios-course/swiftcowboys-team-project/pull/67)

Reviewer: nit:

```suggestion
-  Fixed the implementation for `SleepSetupView` (commited missing changes) - [#67](https://github.com/ios-course/swiftcowboys-team-project/pull/67)
```

Author: thanks, you're right. Changed

-  Created `RoundedRectangleBlue` as `ButtonStyle` - [#70](https://github.com/ios-course/swiftcowboys-team-project/pull/70)

> -  Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)

Reviewer: nit:

```suggestion
-  Update `SwiftFormat` build phases to be able to run linters on `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
```

Author: thanks, changed

-  Updated implementation of `SleepSetupView` to match the mockups. Renamed `EstimatedSleepDurationLabelView` to `MultilineCenteredLabelView` - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
-  Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)

> -  Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target. Removed the `SleepTrackerUITests` target. Unit tests from the `SleepTrackerUITests` target moved into the `SleepTrackerTests` target. Fixed unit tests (to make sure all tests pass) - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Reviewer: This is not important:
 
```suggestion
Removed the `SleepTrackerUITests` target. Unit tests now live in the new `SleepTrackerTests` target. Fixed failed unit tests - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```

Author: fixed

> -  Added an extension `UserDefaults+SharedAppPreferences` for the `UserDefaults` with tests - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)

Reviewer: ```suggestion
-  Added the extension `UserDefaults+SharedAppPreferences` with unit tests - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)
```

Author: fixed

> 

Reviewer 2: Why is there an empty line here?

Author: thanks, didn't pay attention. Fixed

> -  Added extension `Date+ShortComponents`for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Reviewer 3: ```suggestion
-  Added extension `Date+ShortComponents`for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/87)
```
Incorrect reference to PR

Author: thanks, fixed

> -  Added extension `Date+ShortComponents`for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Reviewer: nit:

I know from the name the extension is for the `Date` :) 

```suggestion
-  Added the `Date+ShortComponents` extension - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```

Author: yeah, you're right :)

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added

-  Created initial Xcode project with `MVVM` folder architecture - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added `SwiftFormat` with rules as part of the build phase - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added `SwiftLint` with rules as part of the build phase - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added `.gitignore` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added this `CHANGELOG.md` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added an empty `README.md` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
-  Added images from `Figma` to the assets - [#33](https://github.com/ios-course/swiftcowboys-team-project/pull/33)
-  Added `AppScheme` enum (with tests) according to the project color scheme - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)

> -  Added `Color` extension - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)

Author: done

-  Added `RoundedRectangleBlue` as `ButtonStyle` - [#70](https://github.com/ios-course/swiftcowboys-team-project/pull/70)
-  Added an extension for the `UserDefaults` - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)
-  Added extension for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

### Changed

- Filled the `README` with the description and "Authors" section - [#31](https://github.com/ios-course/swiftcowboys-team-project/pull/31)
- Removed all images with 1x scale from project assets - [#61](https://github.com/ios-course/swiftcowboys-team-project/pull/61)
- Removed spaces from all project folder/file names - [#62](https://github.com/ios-course/swiftcowboys-team-project/pull/62)
- Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
- Updated implementation of `SleepSetupView` to match the mockups - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
- Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)

> - Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Author: @dersim-davaod why is it wrong? I can't understand, can u explain, please?

> - Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Reviewer: The description in the changelog doesn't match the changes introduced in ht PR.

- Removed the `SleepTrackerUITests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

> - Added extension for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)

Reviewer 2: Why is this line in the "changed" section?

Author: You're right, thanks. Moved

### Fixed

- Fixed the bug after merging when initial information appeared in the`README.md` after the merging - [#32](https://github.com/ios-course/swiftcowboys-team-project/pull/32/files)

> 

Author: fixed
